Userland/Libraries/LibWeb/WebIDL/ObservableArray.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Runtime/Array.h>
+#include <LibWeb/WebIDL/ExceptionOr.h>
+
+namespace Web::WebIDL {
+
+// https://webidl.spec.whatwg.org/#idl-observable-array
+class ObservableArray : public JS::Array {
+public:
+    static JS::NonnullGCPtr<ObservableArray> create(JS::Realm& realm);
+
+    virtual JS::ThrowCompletionOr<bool> internal_set(JS::PropertyKey const& property_key, JS::Value value, JS::Value receiver, JS::CacheablePropertyMetadata* metadata = nullptr) override;
+    virtual JS::ThrowCompletionOr<bool> internal_delete(JS::PropertyKey const& property_key) override;
+
+    using SetAnIndexedValueCallbackFunction = Function<ExceptionOr<void>(JS::Value&)>;
+    using DeleteAnIndexedValueCallbackFunction = Function<ExceptionOr<void>()>;
+
+    void add_on_set_an_indexed_value_callback(SetAnIndexedValueCallbackFunction&& callback);
+    void add_on_delete_an_indexed_value_callback(DeleteAnIndexedValueCallbackFunction&& callback);
```

Comment:
These are more `set` than `add` right? Since it's not possible to register multiple callbacks.